### DISCUSSION TITLE
can not use from_pkl function

### DISCUSSION THREAD

**Discussion by Lili-Cao** [chemosim-lab/ProLIF#200] [Q&A]
Hi, 

I was trying to use the from_pkl function to get the interaction fps saved as a pkl file. But I encountered an error.
Below shows how I generate the ifp, save the file and read the file

Could you please tell me what is the problem? Thank you!

fp = plf.Fingerprint()
ifp = plf.Fingerprint(fp.interactions)
ifp.run_from_iterable([ref_mol], protein_mol)
ifp.to_pickle(path=Path(ligdir)/"ifps"/(Path(name).with_suffix(".pkl"))

however, I cannot use 
plf.Fingerprint.from_pkl('path_to_pkl.pkl')
to get the fps,
The error message I get was from 
/dill/_dill.py:  
obj = StockUnpickler.load(self)
UnpicklingError: state is not a dictionary

**Comment by cbouy** [comment]
I'll need a bit more info: which version and platform are you using, are you defining custom interactions? Also are you running this from a plain python script or from a jupyter/colab notebook?

**Reply by Lili-Cao** [reply]
Hi,

I am using version '2.0.3', running it in a jupyternote book in vscode.
I didn't define a custom interaction, just used the default ones.
Here are the related codes in my case

for index, name in enumerate(sys):
    print (name)
    records[index]=name
    reflig_file = Path(ligdir)/Path(name).with_suffix(".pdb")
    protein_file = Path(prodir)/Path(name).with_suffix(".pdb")
    print ("reflig_file", reflig_file)
    print ("protein_file", protein_file)
    u1 = mda.Universe(reflig_file.as_posix(), guess_bonds=True)
    ref_mol = plf.Molecule.from_mda(u1)
    u2 = mda.Universe(protein_file.as_posix(), guess_bonds=True)
    protein_mol = plf.Molecule.from_mda(u2)

    fp = plf.Fingerprint()
    ifp = plf.Fingerprint(fp.interactions)
    ifp.run_from_iterable([ref_mol], protein_mol)
    ifp.to_pickle(path=Path(ligdir)/"ifps"/(Path(name).with_suffix(".pkl")))

**Reply by cbouy** [reply]
I can't reproduce your issue, could you try to run your script with the tutorial files?
You should only need to switch these:
```python
reflig_file = plf.datafiles.datapath / "vina" / "lig.pdb"
protein_file = plf.datafiles.datapath / "vina" / "rec.pdb"
```

Also which OS are you using?

And as a side-note, this line `ifp = plf.Fingerprint(fp.interactions)` is not necessary, you can directly use the `fp` object instead of `ifp`
